halo2_proofs/src/dev.rs

```diff
@@ -809,7 +838,7 @@ impl<F: FieldExt> MockProver<F> {
                         {
                             move |index, _, _| {
                                 let (column, at) = &queries[index];
-                                let resolved_row = (row + at.0) % n;
+                                let resolved_row = (row + n + at.0) % n;
```

Comment:
I can't really see the difference between `(row + n + at.0) % n` and `(row + at.0) % n`. Probably I'm missing smtg.

Reply:
reason for this is that when row is small and at is negative, `(row + at.0) % n` may be negative? say row is 1, and at is -3, so a negative index is calculated here. Is it correct?

Reply:
That makes sense. But it could happen the same before no? Like in the previous status of that, there was the same possibility.

Reply:
previous codes also includes this logic, but ealier.  <https://github.com/appliedzkp/halo2/pull/38/files#diff-faaf1a95220be5b9b73f935e6659140eac1bd000fbbccb6e102589b4c3ec49c5L802>

Reply:
I understand thanks!